### ISSUE TITLE
Dependency hell

### ISSUE THREAD

**Issue by spolischook** [sonata-project/sandbox#602]
I was try ```composer install``` on both master and 2.3 and came to dependency hell.
From master:
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - The requested package liip/imagine-bundle dev-master exists as liip/imagine-bundle[0.x-dev, 1.0.0, 1.0.0-alpha1, 1.0.0-alpha2, 1.0.0-alpha3, 1.0.0-alpha4, 1.0.0-alpha5, 1.0.0-alpha6, 1.0.0-alpha7, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.0.5, 1.0.6, 1.0.7, 1.0.8, 1.0.x-dev, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.2.5, 1.2.6, 1.2.7, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.4.0, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.6.0, 1.7.0, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.8.0, 2.0.x-dev, v0.10.0, v0.10.1, v0.11.0, v0.11.1, v0.12.0, v0.13.0, v0.14.0, v0.15.0, v0.15.1, v0.16.0, v0.17.0, v0.17.1, v0.18.0, v0.19.0, v0.20.0, v0.20.1, v0.20.2, v0.21.0, v0.21.1, v0.9.0, v0.9.1, v0.9.2, v0.9.3, v0.9.4] but these are rejected by your constraint.
  Problem 2
    - sonata-project/media-bundle 3.x-dev requires kriswallsmith/buzz ^0.15 -> satisfiable by kriswallsmith/buzz[v0.15] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.5.1 requires knplabs/gaufrette ^0.1.6 || ^0.2 || ^0.3 -> satisfiable by knplabs/gaufrette[0.2.0, 0.2.1, v0.1.6, v0.1.7, v0.1.8, v0.1.9, v0.3, v0.3.1] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.5.0 requires knplabs/gaufrette ^0.1.6 || ^0.2 -> satisfiable by knplabs/gaufrette[0.2.0, 0.2.1, v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.4.0 requires knplabs/gaufrette ^0.1.6 || ^0.2 -> satisfiable by knplabs/gaufrette[0.2.0, 0.2.1, v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.3.1 requires knplabs/gaufrette ^0.1.6 || ^0.2 -> satisfiable by knplabs/gaufrette[0.2.0, 0.2.1, v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.3.0 requires knplabs/gaufrette ^0.1.6 || ^0.2 -> satisfiable by knplabs/gaufrette[0.2.0, 0.2.1, v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.2.0 requires knplabs/gaufrette ^0.1.6 || ^0.2 -> satisfiable by knplabs/gaufrette[0.2.0, 0.2.1, v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.1.0 requires knplabs/gaufrette ^0.1.6 -> satisfiable by knplabs/gaufrette[v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these conflict with your requirements or minimum-stability.
    - sonata-project/media-bundle 3.0.0 requires knplabs/gaufrette ^0.1.6 -> satisfiable by knplabs/gaufrette[v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these conflict with your requirements or minimum-stability.
    - sonata-project/news-bundle dev-master requires sonata-project/media-bundle ^3.0 -> satisfiable by sonata-project/media-bundle[3.0.0, 3.1.0, 3.2.0, 3.3.0, 3.3.1, 3.4.0, 3.5.0, 3.5.1, 3.x-dev].
    - Installation request for sonata-project/news-bundle dev-master -> satisfiable by sonata-project/news-bundle[dev-master].
```

**Comment by spolischook** [comment]
ok,  now I see https://travis-ci.org/sonata-project/sandbox/branches
@rande dose I can help with that?

**Comment by jordisala1991** [comment]
Can you try using `2.4-develop` branch?

**Comment by spolischook** [comment]
Yep, it's works, expect this ugly error message at the end
![Error](http://storage1.static.itmages.com/i/17/0529/h_1496051961_4627256_543359d7d1.png)

**Comment by spolischook** [comment]
Could not find any status check on this branch #594 
There is no any travis build to check at list the installation is working?

**Comment by AntonioCS** [comment]
I just ran _composer install_ on _sandbox-2.3_ and got this:

```
root@2b38bb918ed8:/application# composer install
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - The requested package sonata-project/ecommerce dev-develop-deprecated exists as sonata-project/ecommerce[0.0.1, 1.0.0, 1.x-dev, 2.0.0, 2.1.0, 2.1.1, 2.2.0, 2.x-dev, dev-master, 3.x-dev] but these are rejected by your constraint.

```

Not as severe as what was detailed on the initial issue but still bad.
Is there any good solution for this?

**Comment by AntonioCS** [comment]
I tried resolving this by changing composer.json but I just keep getting different errors:
```

root@2b38bb918ed8:/application# composer install
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for sonata-project/ecommerce 3.x-dev -> satisfiable by sonata-project/ecommerce[3.x-dev].
    - sonata-project/ecommerce 3.x-dev requires sonata-project/classification-bundle ^3.3 -> satisfiable by sonata-project/classification-bundle[3.3.0, 3.3.1, 3.3.2, 3.4.0, 3.5.0, 3.x-dev] but these conflict with your requirements or minimum-stability.

root@2b38bb918ed8:/application# composer install
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for sonata-project/ecommerce 2.2.0 -> satisfiable by sonata-project/ecommerce[2.2.0].
    - sonata-project/ecommerce 2.2.0 requires sonata-project/admin-bundle ^3.29 -> satisfiable by sonata-project/admin-bundle[3.29.0, 3.30.0, 3.30.1, 3.x-dev] but these conflict with your requirements or minimum-stability.

root@2b38bb918ed8:/application# 

```

I really wanted to get the sonata-demo running on my machine because it has most of the features I want working and I can just experiment not only by using the features but also by changing the code.

**Comment by AntonioCS** [comment]
After posting my previous comments I noticed the suggestion to use sonata2.4-develop branch.
Here is the result:
```

root@4b16528d1fad:/application# composer install    
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - sonata-project/media-bundle 3.10.0 requires sensio/generator-bundle ^3.1 -> satisfiable by sensio/generator-bundle[3.1.0, 3.1.x-dev, v3.1.1, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7] but these conflict with your requirements or minimum-stability.
    - sonata-project/news-bundle 3.x-dev requires sonata-project/media-bundle ^3.10 -> satisfiable by sonata-project/media-bundle[3.10.0].
    - Installation request for sonata-project/news-bundle 3.x-dev -> satisfiable by sonata-project/news-bundle[3.x-dev].

root@4b16528d1fad:/application# 
```

**Comment by mzampetakis** [comment]
Is there any progress with this issue or it's abandoned?

**Comment by jordisala1991** [comment]
Let's say we need to rework a lot of this dependencies, there could be a lot of things wrongs on that composer.json. If you want to help on this feel free to open a PR and we will help you getting it through. If not, we will get to this project eventually and end up fixing things. :)

**Comment by AntonioCS** [comment]
@jordisala1991 I tried to fix this but I kept getting more and more issues and since I am not super familiar with the project I was not really sure if my changes wouldn't break things even if my changes fixed all the dependencies issues.

Can you clarify:

> we will get to this project eventually and end up fixing things.

When exactly will this be? I just ask because sonatas documentation isn't great but having something running with everything turned on does really help.

**Comment by jordisala1991** [comment]
We dont offer estimations on OSS

**Comment by jordisala1991** [comment]
Dependency hell is solved om the new 3.0 branch.

Closing this.

If you find more issues with the 3.0 branch, please report them in new issues.
